src/containers/Application/ApplicationPageWrapper.tsx

```diff
@@ -111,4 +164,83 @@ function processRedirect(appState: any): void {
   }
 }
 
+interface buildProgressInApplicationProps {
+  elementsState: ApplicationElementStates | undefined
+  responses: ResponsesFullByCode | undefined
+  templateSections: TemplateSectionPayload[]
+  isLinear: boolean
+  currentSection: number
+  currentPage: number
+}
+
+function buildProgressInApplication({
+  elementsState,
+  responses,
+  templateSections,
+  isLinear,
+  currentSection,
+  currentPage,
+}: buildProgressInApplicationProps): ProgressInApplication {
+  if (!elementsState || !responses) return []
+  let sectionsStructure: ProgressInApplication = templateSections.map((section) => {
+    // Create an array with all pages in each section
+    const pages = Array.from(Array(section.totalPages).keys(), (n) => n + 1)
+    const checkIsCurrentPage = (page: number) =>
+      section.index === currentSection && page === currentPage
+    const checkIsPreviousPage = (index: number, page: number) =>
```

Comment:
don't really need the `check` bit in the method naming here, but doesn't hurt to keep

Reply:
I'll remove. : )